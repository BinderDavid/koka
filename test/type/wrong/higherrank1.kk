val ids = [id : forall<a> a -> a]

function wrong(x) {
  choose(x,ids)
}