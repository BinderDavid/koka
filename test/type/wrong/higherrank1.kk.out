test/type/wrong/higherrank1.kk(3,16): error: unannotated parameters cannot be polymorphic
 context : function wrong(x) {
 choose(x,ids)
 }
 term : x
 inferred type: list<forall<a> (a) -> a>
 hint : annotate the parameter with a polymorphic type